components/post_view/post_list/index.jsx

```diff
@@ -0,0 +1,449 @@
+        });
+    }
+
+    getPostsBefore = async (postId) => {
+        this.setLoadingPosts('olderPosts');
+        await this.callLoadPosts(this.props.channelId, postId, PostRequestTypes.BEFORE_ID);
+    }
+
+    getPostsAfter = async (postId) => {
+        this.setLoadingPosts('newerPosts');
+        await this.callLoadPosts(this.props.channelId, postId, PostRequestTypes.AFTER_ID);
+    }
+
+    getUnreadPostsCount = (posts, currentUserId) => {
+        //This can be different than the unreadCount on the sidebar as sytem messages
+        //are not considered for the count.
+        return posts.reduce((count, post) => {
+            if (post.create_at > this.state.lastViewedAt &&
+                post.user_id !== currentUserId &&
+                post.state !== Constants.POST_DELETED) {
+                return count + 1;
+            }
+            return count;
+        }, 0);
+    }
+
+    // used for click on new messages indicator
+    // got to latest if channel is not atEnd.
+    // else scroll to bottom
+    goToLatestPosts = () => {
+        this.storePostsIdsAndClear();
+        this.loadUnreadPosts(this.props.channelId);
+    }
+
+    render() {
+        const posts = this.props.posts || [];
+        let postList;
+        let topRow;
+        let bottomRow;
+        let disableLoadingPosts = false;
+
+        if (posts.length === 0 && this.state.isDoingInitialLoad) {
+            return (
+                <div id='post-list'>
+                    <LoadingScreen
+                        position='absolute'
+                        key='loading'
+                    />
+                </div>
+            );
+        }
+
+        if (this.props.postVisibility >= Constants.MAX_POST_VISIBILITY) {
+            const cannotLoadMoreMessages = (
+                <div className='post-list__loading post-list__loading-search'>
+                    <FormattedMessage
+                        id='posts_view.maxLoaded'
+                        defaultMessage='Looking for a specific message? Try searching for it'
+                    />
+                </div>
+            );
+
+            if (this.state.olderPosts.allLoaded) {
+                bottomRow = cannotLoadMoreMessages;
+            } else if (this.state.newerPosts.allLoaded) {
+                topRow = cannotLoadMoreMessages;
+            } else {
+                topRow = cannotLoadMoreMessages;
+                bottomRow = cannotLoadMoreMessages;
+            }
+
+            disableLoadingPosts = true;
+        }
+
+        if (this.state.olderPosts.allLoaded) {
+            topRow = (
+                <CreateChannelIntroMessage
+                    fullWidth={this.props.fullWidth}
+                />
+            );
+        }
+
+        if (posts.length && !this.state.isDoingInitialLoad) {
+            postList = (
+                <PostList
+                    posts={this.props.posts}
+                    fullWidth={this.props.fullWidth}
+                    actions={{
```

Comment:
This is passing in a new object on each call to render, so it'll cause PostList to re-render whenever PostListWrapper does. To prevent that, you can just pass in `loadOlderPosts` and `loadNewerPosts` as separate props instead of wrapping them in `actions`